app/src/main/java/com/github/sdp_begreen/begreen/firebase/FirebaseDB.kt

```diff
@@ -0,0 +1,162 @@
+        catch (databaseEx: DatabaseException) {
+            Log.d(TAG, "Failed with error message: ${databaseEx.message}")
+            throw databaseEx
+        }
+    }
+
+    /**
+     * Associate a new [value] to the given [key].
+     * Create a new entry if the [key] did not exist
+     *
+     * @param key the key we want to set the value (non empty)
+     * @param value the new value for the [key]
+     */
+    operator fun set(key: String, value: String) {
+        if (key.isNotEmpty()) databaseReference.child(key).setValue(value)
+    }
+
+    /**
+     * Store the given [image] under the ID [userId]
+     * and returns the imageId set for that image in the database
+     * Returns null if the image can't be stored
+     */
+    /**
+     * Adds and [image] for the user [userId] in the database
+     *
+     * @param image the image we want to add to the database
+     * @param userId the ID of the user wanting to store a new image
+     * @return a unique ID under which the image got stored, and null if the image couldn't get stored
+     */
+    suspend fun addImage(image: Bitmap, userId: Int): String? {
+
+        // Points to the node where we want to store the metadata of the image in the realtime database
+        val imageMetadataNode = databaseReference.child("pictures").child("userId")
+            .child(userId.toString())
+        // Generates a new imageID for the image we're going to add, and return null if it fails
+        val uniqueID = imageMetadataNode.push().key ?: return null
+
+        // Points to the node where we want to store the image in Firebase storage
+        val imageNode = storageReference.child("userId").child(userId.toString()).child(uniqueID)
+        // Compress the image in a JPEG format
+        val compressedImage = ByteArrayOutputStream()
+        image.compress(Bitmap.CompressFormat.JPEG, 100, compressedImage)
+        return try {
+            // Upload the image
+            imageNode.putBytes(compressedImage.toByteArray()).await()
+            // Upload the image's metadata (might need to add/remove fields later, and pass them as args)
+            val imageMetadataMap = HashMap<String, Any>()
+            imageMetadataMap["time"] = System.currentTimeMillis()
+            imageMetadataMap["type"] = "plastic bottle"
+
+            imageMetadataNode.child(uniqueID).setValue(imageMetadataMap)
+            uniqueID
+        } catch (e: Error) {
+            null
+        }
+    }
+
+    /**
+     * Retrieves the image associated with the given [userId] and [imageId] from the database
+     *
+     * @param imageId the ID of the image we want to retrieve from the database
+     * @param userId the ID of the user where we should find the image
+     * @return the image, or null if no image was found
+     * @throws StorageException if the image could not get retrieved
+     * @throws DatabaseException if the metadata could not get retrieved
+     *
+     */
+    @SuppressLint("RestrictedApi")

```

Comment:
This does not look like a good idea? https://stackoverflow.com/a/64818775

If there is a good reason to do it, it's exactly what comments are for :)

Reply:
Yes indeed, after discussions David changed it in his PR just after #30